packages/components/src/menu-item/types.d.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { IconType } from '../icon';
+
+export type MenuItemProps = {
```

Comment:
```suggestion
import type { ButtonAsButtonProps } from '../button/types';

export type MenuItemProps = ButtonAsButtonProps & {
```

Since all the rest props are being passed down to the underlying `Button` component, let's add that into the allowed types. I personally think a lot of those `Button` props shouldn't be allowed in this context, but the current implementation allows it, so I guess we can keep it this way for now 🤷 This will also fix the `variant` problem in `ToolsPanel`.

Reply:
`variant` issue solved!